### PR TITLE
Added Fallout 2 port.

### PR DESCRIPTION
**Pull request by orsonmmz**
It has been built from https://github.com/alexbatalov/fallout2-ce (3aea6a9).
I have tested it with RG351M/AmberELEC. It would be great if someone could test gptokeyb mappings with a one analog stick device.

### CODE REVIEW

**Comment by christianhaitian** [comment]
Has this been tested on ArkOS yet?

**Comment by orsonmmz** [comment]
No, I am using AmberELEC exclusively. Is there a specific reason to test it with ArkOS? If so, I will try it out.

**Comment by christianhaitian** [comment]
Portmaster is available for multiple distros and we try to ensure that most ports work across them all.  Since the primary difference between supported distros is buildroot vs Ubuntu/Debian, we try to test one of each.  Confirming functionality on AmberElec means it should work for AmberElec and UnofficialOS on multiple devices.  Confirming functionality on ArkOS means it should work for ArkOS, TheRA, and RetroOZ on multiple devices.  This is given you followed the packaging guidelines documented [here](https://github.com/christianhaitian/PortMaster/blob/main/docs/packaging.md)

**Comment by orsonmmz** [comment]
Got it, I will check it with ArkOS and let you know if it works.

**Comment by orsonmmz** [comment]
You were right to check it with ArkOS. The game starts, but there is something wrong with the graphics - only transitions (fade in/out) are shown correctly, then it switches to text mode where I can still hear game music with some distortions.

Before I dive into debugging, here is the console output:
```
file /dev/mali0 is not of a compatible version (user 10.6, kernel 11.7)
file /dev/mali0 is not of a compatible version (user 10.6, kernel 11.7)
Failed creating base context during opening of kernel driver.
Kernel module may not have been loaded
Failed to create secure directory (/run/user/1002//pulse): No such file or directory
ALSA lib pcm.c:8432:(snd_pcm_recover) underrun occurred
```
If I read it right, there is an incompatibility between the video driver and the userspace library. Is there a known solution to it?
For reference, I am testing ArkOS_RG351MP_v2.0_01152023 and AmberELEC/20220627/Sanshiro Tower.

For the builds, I use an environment created by following [these steps](https://github.com/christianhaitian/rk3326_core_builds/blob/rk3326/docs/chroot.md).

**Comment by christianhaitian** [comment]
Yes that’s an issue I still need to address but it’s a harmless warning.  Just that some libraries are linked against a slight older video driver but they both perform the same.  I get the same flickering issue on a 353 device.  I found if I load the game linking against an older version of sdl, it works.  It doesn’t seem to like the sdl 2.0.20 currently used in ArkOS maybe?  

Change line 33 to the following `LD_PRELOAD=/usr/lib/aarch64-linux-gnu/libSDL2-2.0.so.0.10.0 ./fallout2-ce 2>&1 | tee -a ./log.txt` and try that.

**Comment by orsonmmz** [comment]
Thank you for the advice, it worked like a charm! After applying you hint, I was able to pass the trials of Arroyo (kind of tutorial), which uses most of the game mechanics.

The proposed change does not affect AmberELEC, because libSDL2-2.0.so.0.10.0 does not exist there. Is it ok to keep it as the default setting, or should it be made a conditional one? If so, how I should make it specific to ArkOS?

We are almost there, but there is one thing left: sometimes trying to quit the game via Select+Start prevents accepting any further input. I am still trying to determine the conditions causing it to happen, but maybe you already have a hint? The previous one was bullseye.

**Comment by christianhaitian** [comment]
Hi,

Don't worry about accounting for the sdl2 issue for ArkOS.  I will push out an updated sdl2 for ArkOS soon that will take care of the compatibility for ArkOS.  I will be 2.26.2 so it's in inline with AmberElec.  It seems to have also stabilized the start exit hotkey issue as well on my end.

**Comment by orsonmmz** [comment]
Excellent! I guess there are no more issues to resolve, let me know if you think otherwise.
Thank you for the help and all the effort you put into PortMaster.

**Comment by christianhaitian** [comment]
Merged.  Thank you for your contribution to this project.

**Comment by Smelltastic** [comment]
Hi, sorry to put this here but now that the fallout2 branch is gone I'm not sure where else to put this.

There's an issue with the Fallout2-ce where enemies don't seem to aggro you properly. On the Discord we traced it down to the compiler; kloptops compiled it with the -O3 option and that seemed to work better. There's a similar issue posted on CE's github here: https://github.com/alexbatalov/fallout2-ce/issues/93

kloptops send that recompiled version to me and I've been testing it for a couple days and it's working well. Would it be possible to recompile your version, or replace with kloptops'? He uploaded his here: https://drive.google.com/file/d/142uiy92rHAcMgzNkTuSxWtUOj2C9Oa9A

**Comment by orsonmmz** [comment]
Hi @Smelltastic, good catch, thanks for the report!
I think it is better to go with kloptops binary, as it is already tested by you. I will check it with AmberELEC tonight and given I do not see any other problems, I will post a merge request with the update.

**Comment by orsonmmz** [comment]
It still intrigues me what went wrong here. It turns out that I have already compiled it with -O3, since this is the default for CMAKE_BUILD_TYPE=Release. @kloptops, what compiler have you used?

**Comment by orsonmmz** [comment]
TL;DR: I have tested kloptops binary and the change in NPCs behavior is immediately noticeable, so I have created a MR with the fix. Frankly, I have not paid attention to this, apparently my game reminiscence is already quite corrupted.

@kloptops, I think you are right saying it might be a compiler related issue, here are my findings:

- The first binary uploaded to PortMaster was built from 3aea6a98, now I have updated to 81b93453 - no difference.
- I am using gcc 8.3.0, but neither Release nor MinSizeRel builds changed anything.
- Building with clang-13 fixes the erratic behavior.
- Works ok on PC (commit 81b93453 / gcc 12.2).

**Comment by kloptops** [comment]
I used gcc 10.2.1.
